app/models/active_analytics/views_per_day.rb

```diff
@@ -80,9 +89,25 @@ def height
     end
 
     def self.group_by_site
-      group(:site).pluck("site, SUM(total)").map do |row|
-        Site.new(row[0], row[1])
-      end
+      group(:site, :browser, :device_type, :operating_system)
+        .pluck("site, browser, device_type, operating_system, SUM(total)")
+        .to_a
+        .group_by { |r| r[0] }
+        .map do |site_name, grouped_site|
+          total_sum = 0
+          browsers = Hash.new(0)
+          device_types = Hash.new(0)
+          operating_systems = Hash.new(0)
+
+          grouped_site.each do |row|
+            total_group = row[4].to_i
+            browsers[row[1].to_sym] += total_group if row[1]
+            device_types[row[2].to_sym] += total_group if row[2]
+            operating_systems[row[3].to_sym] += total_group if row[3]
+            total_sum += total_group
+          end
```

Comment:
The aggregation has to live in Ruby for the moment because I didn't want to hard code which exact values for `browser`, `operating_system` and `device_type` we want to allow. Technically this would be currently know from my implementation of `user_agent_extractors`.

But it didn't feel right to add / generate mildly complex SQL to aggregate exact values when not knowing whether this might just be good enough.